File: backend/src/main/java/com/se114p12/backend/entities/promotion/Promotion.java
package com.se114p12.backend.entities.promotion;

import com.se114p12.backend.entities.BaseEntity;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.math.BigDecimal;
import java.time.Instant;

@Data
@EqualsAndHashCode(callSuper = true)
@Entity
@Table(name = "promotions")
public class Promotion extends BaseEntity {

    private String description;

    @NotNull
    private BigDecimal discountValue;

    @NotNull
    private BigDecimal minValue;

    @NotNull
    private Instant startDate;

    @NotNull
    private Instant endDate;

    private String code;

    // true: toàn bộ người dùng, false: chỉ định người dùng
    private Boolean isPublic;
}

File: frontend/MAM/app/src/main/java/com/example/mam/entity/Promotion.kt
package com.example.mam.entity

import java.text.DecimalFormat
import java.time.Instant

data class Promotion(
    val code: String = "",
    val value: Int = 0,
    val name: String = "",
    val description: String = "",
    val startDate: Instant = Instant.now(),
    val endDate: Instant = Instant.now(),
    val minValue: Int = 0,
    val createAt: Instant = Instant.now(),
    val updateAt: Instant = Instant.now(),
){
    fun getValueToString(): String {
        val formatter = DecimalFormat("#,###")
        return "- ${formatter.format(value)} VND"
    }
}
